idaes/core/util/tables.py

```diff
@@ -234,8 +235,19 @@ def create_stream_table_dataframe(
             for i in disp_dict[k]:
                 if i is None:
                     stream_attributes[key][k] = value(disp_dict[k][i])
+                    if k not in full_keys:
+                        full_keys.append(k)
                 else:
                     stream_attributes[key][k + " " + str(i)] = value(disp_dict[k][i])
+                    if k + " " + str(i) not in full_keys:
+                        full_keys.append(k + " " + str(i))
+
+    # Check for missing rows in any stream, and fill with "-" if needed
+    for k, v in stream_attributes.items():
+        for r in full_keys:
+            if r not in v.keys():
+                # Missing row, fill with placeholder
+                v[r] = "-"
 
     return DataFrame.from_dict(stream_attributes, orient=orient)
```

Comment:
Try `return DataFrame.from_dict(stream_attributes, orient=orient).round(6)` that should round the whole dataframe to 6 sig figs

Reply:
I tried something similar, but didn't see any difference in the json. I will try again when I have some time.

Reply:
Okay this is very weird. It's the stream table with the rounding issue but the dataframe we use to create the stream table has the `round` function applied.

Let me look into this more.

Reply:
Ah okay, pandas will only perform the `round` on columns that are numerical types. Since you replaced the NaNs with `-` it changed the column type from `float64` to `object`.

On line 395 (`.round(self._sig_figs)`) and line 422 (`c = self._serialized_contents[unit_name][content_type].round(self._sig_figs).to_dict("index")`) replace the `.round(self._sig_figs)` with
```
.applymap(lambda x: round(x, self._sig_figs) if isinstance(x, (int, float)) else x)
```

That will apply a round to the numbers and leave the strings alone

Reply:
I have implemented the suggestions above, and we will see if the tests pass now.